### PR TITLE
Add missing plugin command to help

### PR DESCRIPTION
**Pull request by p8**
### Summary
Because the PluginCommand is defined with `hide_command!`, running
`bin/rails -h` does not mention the plugin command.

As an alternative to removing the `hide_command!` this adds the plugin
command to the USAGE. Unless it's an engine, because adding an engine to
an engine doesn't make sense.

Another option could be to remove the `hide_command!` from PluginCommand.

Both the `plugin` and `engine` terms are used so a user can search for both.

image

### CODE REVIEW

**Comment by zzak** [comment]
Go away rails-bot

File: railties/lib/rails/commands/help/USAGE [hunk 1]
```diff
@@ -9,6 +9,7 @@ The most common rails commands are:
 <% unless engine? -%>
  new          Create a new Rails application. "rails new my_app" creates a
               new application called MyApp in "./my_app"
+ plugin       Create a new Rails engine
```
Comment (jonathanhefner) [inline railties/lib/rails/commands/help/USAGE:12]: Technically, `rails plugin new my_plugin` creates a simple railtie, while `rails plugin new my_plugin --mountable` or `--full` creates an engine.  So, perhaps...

```suggestion
 plugin       Create a new Rails railtie or engine
```

Comment (p8) [reply railties/lib/rails/commands/help/USAGE:12]: `rails plugin new -h | grep railtie` doesn't return any results.
The guides also don't seem to mention `Railtie` https://guides.rubyonrails.org/plugins.html
What do you think about...
```suggestion
 plugin       Create a new Rails plugin or engine
```

Comment (jonathanhefner) [reply railties/lib/rails/commands/help/USAGE:12]: I feel like "plugin" means "railtie or engine".  So if we prefer to not mention "railtie" explicitly, I think "Create a new Rails plugin" would suffice.  Although I should point out that `rails plugin new -h | grep engine` returns only the engine-specific options, and https://guides.rubyonrails.org/plugins.html doesn't mention "engine" at all (though it does hint at railtie, e.g. `require "yaffle/railtie"`).

Something else I just thought of -- should this be `plugin new`?

```suggestion
 plugin new   Create a new Rails engine
```

`rails plugin my_plugin` seems to be equivalent to `rails plugin new -h`.  (I wonder if we can / should change that? :thinking:)

Comment (p8) [reply railties/lib/rails/commands/help/USAGE:12]: Hmm, good points.

The engine guides does mention `railtie` be describing `bin/rails railties:install:migrations` 
https://guides.rubyonrails.org/engines.html. Plugin does seem the superset of railtie and engine.

The current description of the plugin command also describes it as `rails plugin new`.
```
Description:
    The 'rails plugin new' command creates a skeleton for developing any
    kind of Rails extension with ability to run tests using dummy Rails
    application.
```

Comment (p8) [reply railties/lib/rails/commands/help/USAGE:12]: So my new preference would be:
```suggestion
 plugin new       Create a new Rails railtie or engine
```

**Pull request merged**